crates/walrus-sui/src/client/retry_client/retriable_sui_client.rs

```diff
@@ -134,9 +134,36 @@ impl LazySuiClientBuilder {
 impl LazyClientBuilder<SuiClient> for LazySuiClientBuilder {
     // TODO: WAL-796 Out of concern for consistency, we are disabling the failover mechanism for
     // SuiClient for now.
-    const DEFAULT_MAX_TRIES: usize = 1;
+    const DEFAULT_MAX_TRIES: usize = 5;
```

Comment:
I'm not sure we should bundle all these changes with the actual enabling of the SuiClient's lazy failover, especially since we are not addressing the previously identifie dconsistency issues.

Reply:
Maybe you could turn it on in `#[cfg(msim)]` though?

Reply:
As I mentioned in slack, without this change, any one RPC node creation failure would prevent the `RetriableSuiClient` to fail initialize, also we should just enable Sui client failover as well.